src/Log/LogService.php

```diff
@@ -0,0 +1,135 @@
+ */
+
+namespace Opus\Log;
+
+class LogService
+{
+
+	public static $logfile;
+	public static $logger;
+
+	public static function getInstance($logFileName){
+
+	    $registry = \Zend_Registry::getInstance();
+	    $config = $registry->get('Zend_Config');
+
+        $logfilePath = $config->workspacePath . '/log/' . $logFileName;
+        self::$logfile = @fopen($logfilePath, 'a', false);
+        $logfile = self::$logfile;
+        if ($logfile === false) {
+            $path = dirname($logfilePath);
+
+            if (! is_dir($path)) {
+                throw new Exception('Directory for logging does not exist');
+            } else {
+                throw new Exception('Failed to open logging file:' . $logfilePath);
+            }
+        }
+
+        $logService = new static();
+        return $logService;
+    }
+
+    public function setLog($logLevelName, $format = null){
+
+        $registry = \Zend_Registry::getInstance();
+        $config = $registry->get('Zend_Config');
+
+        $GLOBALS['id_string'] = uniqid(); // Write ID string to global variables, so we can identify/match individual runs.
+
+        if(isset($config->logging->log->translation->file->format)){
+            $format = $config->logging->log->translation->file->format;
+        } elseif ($format == null){
+            $format = '%timestamp% %priorityName% (%priority%, ID '.$GLOBALS['id_string'].'): %message%' . PHP_EOL;
+        }
+        $formatter = new \Zend_Log_Formatter_Simple($format);
+
+        $writer = new \Zend_Log_Writer_Stream(self::$logfile);
+        $writer->setFormatter($formatter);
+
+        $logger = new \Zend_Log($writer);
+//        $logLevelName = 'INFO';
+        $logLevelNotConfigured = false;
+
+         if (isset($config->logging->log->level)) {
+             $logLevelName = strtoupper($config->logging->log->level);
+         } else {
+             $logLevelNotConfigured = true;
+         }
+
+        $zendLogRefl = new \ReflectionClass('Zend_Log');
+
+        $invalidLogLevel = false;
+
+        $logLevel = $zendLogRefl->getConstant($logLevelName);
+
+        if (empty($logLevel)) {
+            $logLevel = Zend_Log::INFO;
+            $invalidLogLevel = true;
+        }
+
+        // filter log output
+        $priorityFilter = new \Zend_Log_Filter_Priority($logLevel);
+        \Zend_Registry::set('LOG_LEVEL', $logLevel);
+        $logger->addFilter($priorityFilter);
+
+        if ($logLevelNotConfigured) {
+            $logger->warn('Log level not configured, using \'' . $logLevelName . '\'.');
+        }
+
+        if ($invalidLogLevel) {
+            $logger->err('Invalid log level \'' . $logLevelName .
+                '\' configured.');
+        }
+
+        \Zend_Registry::set('Zend_Log', $logger);
+
+        $logger->debug('Logging initialized');
+        self::$logger = $logger;
+        return $logger;
+    }
+
+	public function getLog($logLevelName){      //not sure what exactly to return here
```

Comment:
We do not distinguish between logs by Log level. We have multiple logs that may have different logs levels but we use a name to get the logger we want, e.g. 'default', 'doi', 'translation', ...

The function should check if a logger with that name already exists. If not it should try to create a logger from the configuration. If that fails it should throw an exception, UnknownLogger. Modules that want to use a new custom 
logger would add the createLog call to their Bootstrap process. At least that is what I am thinking right now, until we run into problems.

Reply:
When I am saying creating a logger from "configuration" it is something that could be added in the next step. It does not block any of the other changes request here. It would be fine to have a function that creates the default logger for now as long as other loggers can be added dynamically. The next step then will be to define multiple loggers in the configuration as outlined in the ticket. Those loggers can just be obtained by their name and will be created by the LogService class if necessary. Configured will be all loggers that we know about at this point. However the system should not be closed to additional loggers added from the outside.